Adding more context to upgrading process documentation

st13g opened the pull request:
### Description

This PR add more context to the upgrading process, and is needed in order to give a clear distinctions between upgrading packages and upgrading releases.

### Related Issue
This PR closes #266 .

### Checklist
- [x] I have read and followed the [Ubuntu Server contributing guide](https://documentation.ubuntu.com/server/contributing/).
- [x] I have signed the [Contributor License Agreement (CLA)](https://ubuntu.com/legal/contributors).
- [x] My pull request is linked to an existing issue (if applicable).
- [x] I have tested my changes, and they work as expected.

---

if there is something to be fixed, or to be improved please feel free to let me know!!

TheJJ commented on how-to/software/upgrade-your-release.md:
````diff
@@ -22,6 +36,7 @@ To ensure a successful upgrade, review the following items:
   ```bash
   sudo apt update
   sudo apt upgrade
+  sudo reboot
````
recommending a reboot after every upgrade is not something we should do. rather we could hint at "needrestart" and needs-reboot status after kernel upgrades

panlinux replied:
This is not for every upgrade, but for the upgrade prior to the *release* upgrade. Still, I agree that a reboot should only be done if requested by the system. And if I'm not mistaken, `do-release-upgrade` will tell you about it if needed.
So I think replace the reboot command with a note that depending on the packages that were upgrade, you might have to reboot the system if asked to do so.

TheJJ commented on how-to/software/upgrade-your-release.md:
```diff
@@ -7,10 +7,24 @@ We recommend running a Long Term Support (LTS) release as it provides 5 years of
+| :--- | :---: |
+| apt update | Updates packages index (metadata) | 
+| apt upgrade | Upgrades only installed packages within the current distribution |
+| do-release-upgrade | Upgrades the entire OS to next release |
```
while explaining these differences is not a bad idea, I'd rather reformulate or clarify the sections (pre-upgrade checklist, upgrade the system).

TheJJ replied:
sorry, didn't see that in your bug, alright then :)

TheJJ reviewed (changes requested):
thanks for the additions!
i think updating the document in whole makes more sense than adding a new section about upgrade differences.

panlinux commented on how-to/software/upgrade-your-release.md:
````diff
@@ -7,10 +7,24 @@ We recommend running a Long Term Support (LTS) release as it provides 5 years of
 
+## Update vs Upgrade vs Release-Upgrade
+
+```{note}
````
I think this can be a normal paragraph, and not a note. It looks better, since the text follows a new title.

panlinux commented:
No need to submit another PR, @st13g , please use this one.

st13g commented:
Hi @panlinux @TheJJ i've submitted new commits, please let me know your thoughts, any feedback is always welcome!

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -1,41 +1,50 @@
+* **Review Releases notes:** Always check the **release notes** for the new Ubuntu version we are moving to. This can be found on the [Ubuntu Wiki Releases Page](https://wiki.ubuntu.com/Releases).
 
-* Fully update the system. The upgrade process works best when the current system has all the latest updates installed. You should confirm that these commands complete successfully and that no further updates are available. We also suggest rebooting the system after all the updates are applied, to ensure the latest kernel is being run. To upgrade, run the following commands:
+* **Fully update the current system:** The release upgrade process works best when the current system has all the latest updates installed. This is a standard **package upgrade**, **apt update** will refresh package lists (sync our package index with the upstream sources), and **apt upgrade** will install the latest versions of installed packages.
```
Replace:

This is a standard **package upgrade**, **apt update** will refresh package lists (sync our package index with the upstream sources), and **apt upgrade** will install the latest versions of installed packages.

With:

This is a standard **package upgrade**: **apt update** will refresh the package index database, and **apt upgrade** will download and install the latest versions of installed packages.

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -1,41 +1,50 @@
+  2. Confirm both commands complete successfully and no further updates are available.
 
-* The upgrade process takes time to complete. You should have dedicated time to participate in the upgrade process.
+  3. After applying all updates, **reboot your system** to ensure the latest kernel is running.
```
Replace:

   3. After applying all updates, **reboot your system** to ensure the latest kernel is running.
 
With:

 3. After applying all updates, it may be necessary to **reboot your system**. The release upgrade process will let you know if that's needed, but you can also check manually before: if the file `/run/reboot-required` exists, then you will need to reboot.

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -1,41 +1,50 @@
+* **Check that there is enough free space:** A release upgrade involves downloading hundreds of new packages, which can be several gigabytes. Make sure you have enough free disk space.
 
-* Backup all your data. Although upgrades are normally safe, there is always a chance that something could go wrong. It is extremely important that the data is safely copied to a backup location to allow restoration if there are any problems during the upgrade process.
+* **Dedicate time for the upgrade:** This process takes time. So we should dedicated period during which we can monitor the upgrade and respond to any prompts.
```
Replace:

This process takes time. So we should dedicated period during which we can monitor the upgrade and respond to any prompts.

With:

This is an interactive process. The release upgrade will sometimes stop and ask questions, so you should monitor the upgrade and be available to respond.

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -1,41 +1,50 @@
-* Backup all your data. Although upgrades are normally safe, there is always a chance that something could go wrong. It is extremely important that the data is safely copied to a backup location to allow restoration if there are any problems during the upgrade process.
+* **Dedicate time for the upgrade:** This process takes time. So we should dedicated period during which we can monitor the upgrade and respond to any prompts.
+
+* **Understand Third-party repositories:**  Third-party software repositories and Personal Package Archives (PPAs) are disabled during upgrade. While software installed from these sources will not be removed, it's the most common cause of upgrade issues. Be prepared to re-enable them or find updated versions after the upgrade.
```
Replace:

are disabled during upgrade.

With:

are disabled during the release upgrade.

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -1,41 +1,50 @@
+
+* **Understand Third-party repositories:**  Third-party software repositories and Personal Package Archives (PPAs) are disabled during upgrade. While software installed from these sources will not be removed, it's the most common cause of upgrade issues. Be prepared to re-enable them or find updated versions after the upgrade.
+
+* **Backup all your data:**  Although upgrade are normally safe, there is always a chance that something could go wrong. It is extremely important that the data is safely copied to a backup location to allow restoration if any problems occur.
```
Typo: replace `Although upgrade are` with `Although upgrades are`

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -1,41 +1,50 @@
 
 ## Upgrade the system
 
+
```
Unnecessary extra blank line.

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -1,41 +1,50 @@
 We recommend upgrading the system using the `do-release-upgrade` command on Server edition and cloud images. This command can handle system configuration changes that are sometimes needed between releases. To begin the process, run the following command:
 
+To start the process, run this command:
+
```
You have to remove the "To begin the process, run the following command:" from the previous paragraph, or else these are two prompts.

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -89,9 +98,9 @@ Configuration file '/etc/ssh/ssh_config'
+### Removing Obsolete Packages
 
-After all packages are updated, you can choose to remove any obsolete, no-longer-needed packages:
+After all packages are updated, you can choose to remove any obsolete packages that are  no longer needed packages:
```
Replace:

After all packages are updated, you can choose to remove any obsolete packages that are  no longer needed packages:

With:

After all packages are updated, you can choose to remove any obsolete packages that are  no longer needed:

panlinux reviewed (changes requested):
Thanks for the updates! Here is my review pass.

st13g commented:
Hi @panlinux , I really appreciate your time and feedback, I've submitted a new commit, addressing all the feedback provided, please if there is something else to be improved, do not hesitate to let me know, Thanks !! :D

panlinux commented on how-to/software/upgrade-your-release.md:
```diff
@@ -89,9 +97,9 @@ Configuration file '/etc/ssh/ssh_config'
+### Removing Obsolete Packages
 
-After all packages are updated, you can choose to remove any obsolete, no-longer-needed packages:
+After all packages are updated, you can choose to remove any obsolete packages that are  no longer needed:
```
Two spaces between "are" and "no": "are  no". Should be one.

panlinux commented:
Thanks @st13g , just minor things now. We are close!

st13g commented:
Hi @panlinux Ive taken care of all your comments, please feel free to let me know if there something else that need to be changed. Also, thanks a lot for your patience and support. :D

TheJJ reviewed (approved):
thank you for your contribution and patience! this is now ready to ship!
